compiler/rustc_ast_lowering/src/contract.rs

```diff
@@ -0,0 +1,260 @@
+use crate::LoweringContext;
+
+impl<'a, 'hir> LoweringContext<'a, 'hir> {
+    pub(super) fn lower_contract(
+        &mut self,
+        body: impl FnOnce(&mut Self) -> rustc_hir::Expr<'hir>,
+        contract: &rustc_ast::FnContract,
+    ) -> rustc_hir::Expr<'hir> {
+        match (&contract.requires, &contract.ensures) {
+            (Some(req), Some(ens)) => {
+                // Lower the fn contract, which turns:
+                //
+                // { body }
+                //
+                // into:
+                //
+                // {
+                //      let __postcond = if contracts_checks() {
```

Comment:
The suggestion is for this code to be either:
```rust
let __postcond = if false {
  // ...
}
```
or 
```rust
let __postcond = if true {
  // ...
}
```
depending on the value of the `-Z contract-checks`

Reply:
The reason for this to be an intrinsic is for tools / custom backends to be able to override the behavior

Reply:
Why would you still need the lang item?